src/experimental-bots/spotFiller/spotFillerMultithreaded.ts

```diff
@@ -2068,6 +2081,18 @@ export class SpotFillerMultithreaded {
 			};
 		}
 
+		if (Array.isArray(nodeFilled)) {
+			nodeFilled = nodeFilled[0];
+		}
```

Comment:
same thing here-- we are coercing the type to nodeFilled, so this should never happen. if it's not an instance of NodeToFill, we should throw an error